### DISCUSSION TITLE
Environment Variable question

### DISCUSSION THREAD

**Discussion by Hexalogy** [mbround18/valheim-docker#490] [Help!] [answered]
Can someone help me understand why the environment variable set in the `docker-compose.yml` doesn't reflect in the `start_server.sh` found in `/home/steam/valheim` inside the running container?

docker-compose.yml:
![image](https://user-images.githubusercontent.com/9726028/145479468-9cde7a6f-d2be-4660-a7bf-c9170e5caabe.png)

start_server.sh:
![image](https://user-images.githubusercontent.com/9726028/145479734-38d88540-773e-4508-840d-d025b7d1b321.png)

If I understand correctly, start_server.sh is the final step in launching the server.

**Comment by mbround18** [comment, marked as answer]
@Hexalogy we don't actually use the provided scripts to run the server, instead we wrap executing the server in a tool system to provide the ability to add those custom jobs such as auto backups, updates, and mod installations. Technically we could use shell scripts to launch and maintain the server but, as one of the devs of this toolkit, it was easier and safer to build it in Rust. This gave us low level access to anything we need outside the scope of shell in order to create a secure, fast, fluid, and performant end user experience. 


If you find a functionality missing from the included toolkit, we also take suggestions! Please feel free to toss something up in the ideas section of the discussion! 


If you are interested more in the inner workings of the included toolkit, please feel free to ask away <3 


However, if you are seeking a more shell oriented experience, there is a shell friendly docker container for Valheim that can be found at [lloesche/valheim-server-docker](https://github.com/lloesche/valheim-server-docker)
